core/server/master/src/test/java/alluxio/master/file/FileSystemMasterFsOptsTest.java

```diff
@@ -294,246 +292,161 @@ public void deleteDirRecursiveNoPermOnFile() throws Exception {
+          DeleteContext.mergeFrom(DeletePOptions.newBuilder().setRecursive(true)));
+      fail("Deleting a directory without permission on current directory should fail");
+    } catch (AccessControlException e) {
+      assertTrue(e.getMessage().contains("access=rwx, path=/nested/test/dir: failed at dir"));
     }
-    // The existing files/dirs will be: /, /nested/, /nested/nested/,
-    // /nested/nested/test/, /nested/test/file
-    assertNotEquals(IdUtils.INVALID_FILE_ID, mFileSystemMaster.getFileId(ROOT_URI));
-    assertNotEquals(IdUtils.INVALID_FILE_ID,
-        mFileSystemMaster.getFileId(new AlluxioURI("/nested")));
-    assertNotEquals(IdUtils.INVALID_FILE_ID,
-        mFileSystemMaster.getFileId(new AlluxioURI("/nested/nested")));
-    assertNotEquals(IdUtils.INVALID_FILE_ID,
-        mFileSystemMaster.getFileId(new AlluxioURI("/nested/nested/test")));
-    assertNotEquals(IdUtils.INVALID_FILE_ID,
-        mFileSystemMaster.getFileId(new AlluxioURI("/nested/nested/test/file")));
+
+    assertNotEquals(IdUtils.INVALID_FILE_ID, mFileSystemMaster.getFileId(NESTED_DIR_URI));
+    assertNotEquals(IdUtils.INVALID_FILE_ID, mFileSystemMaster.getFileId(NESTED_DIR_FILE_URI));
     // The other files should be deleted successfully
     assertEquals(IdUtils.INVALID_FILE_ID,
-        mFileSystemMaster.getFileId(new AlluxioURI("/nested/nested/test/file2")));
+        mFileSystemMaster.getFileId(new AlluxioURI("/nested/test2/dir")));
+    assertEquals(IdUtils.INVALID_FILE_ID,
+        mFileSystemMaster.getFileId(new AlluxioURI("/nested/test2/dir/file")));
   }
 
   @Test
-  public void deleteDirRecursiveNoPermOnDir() throws Exception {
+  public void deleteDirRecursiveDiffPerOnChildDir() throws Exception {
     // The structure looks like below
     // /nested/
     //
     // /nested/test/
-    // /nested/test/file
-    // /nested/test/file2
     // /nested/test/dir
     // /nested/test/dir/file
     //
     // /nested/test2/
-    // /nested/test2/file
-    // /nested/test2/file2
     // /nested/test2/dir
     // /nested/test2/dir/file
-    // userA has no permission on /nested/test/file
+    // userA has no permission on /nested/test
     // So deleting the root will fail on:
-    // /nested/, /nested/test/, /nested/test/file
-    createFileWithSingleBlock(NESTED_FILE_URI);
-    createFileWithSingleBlock(NESTED_FILE2_URI);
+    // /nested/test/dir/, /nested/test/dir/file
     createFileWithSingleBlock(NESTED_DIR_FILE_URI);
-    createFileWithSingleBlock(new AlluxioURI("/nested/test2/file"));
-    createFileWithSingleBlock(new AlluxioURI("/nested/test2/file2"));
     createFileWithSingleBlock(new AlluxioURI("/nested/test2/dir/file"));
 
-    // No permission on the dir, therefore all the files will be kept
-    // although the user has permission over those nested files
     mFileSystemMaster.setAttribute(NESTED_URI, SetAttributeContext
-        .mergeFrom(SetAttributePOptions.newBuilder().setMode(new Mode((short) 0700).toProto())));
```

Comment:
could you please describe why these checks are modified? Better to one by one, then we know whether the deletion is reasonable. And that can also help others understand the justification

Reply:
Test cases are re added to test the permissions of the parent directory, current directory, subdirectory and child file.

Reply:
For permission check rules, refer to the hdfs section in the issue https://github.com/Alluxio/alluxio/issues/16269